### PR TITLE
feat: dispatch the dispatcher

### LINKED ISSUE #11949: Delete dispatch builder
We no longer have any real need for the dispatch builder, and all associated code can be deleted.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #11949

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -246,7 +243,7 @@ public class SwirldsPlatform implements Platform {
     /**
      * All components that need to be started or that have dispatch observers.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:244]: Comment can be updated to not reference dispatch

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 2]
```diff
@@ -370,8 +362,6 @@ public class SwirldsPlatform implements Platform {
 
         registerAddressBookMetrics(metrics, currentAddressBook, selfId);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:364]: Are you intentionally not starting the `recycleBin` anymore?

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:364]: Oops, that's an oversight. We were starting it via introspection, not by contract. I've updated the interface and added it to things to start.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 3]
```diff
@@ -241,7 +241,7 @@ public class SwirldsPlatform implements Platform {
     private final Clearable clearAllPipelines;
 
     /**
-     * All components that need to be started or that have dispatch observers.
+     * All things that need to be started. when the platform is started.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:244]: erroneous `.`

**Review by alittley — DISMISSED** [review]

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java]: fixed

**Review by alittley — APPROVED** [review]

**Review by anthony-swirldslabs — APPROVED** [review]

**Pull request merged**